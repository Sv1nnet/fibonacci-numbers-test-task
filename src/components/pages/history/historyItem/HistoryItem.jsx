import React from 'react';
import PropTypes from 'prop-types';

import './history-item.scss';

const HistoryItem = ({ number, result }) => (
  <tr>
    <th>{number}</th>
    <th>{result}</th>
  </tr>
);

HistoryItem.propTypes = {
  number: PropTypes.number.isRequired,
  result: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
  ]).isRequired,
};

export default HistoryItem;
