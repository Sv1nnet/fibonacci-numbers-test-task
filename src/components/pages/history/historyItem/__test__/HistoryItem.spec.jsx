import {
  it,
  describe,
  expect,
} from '@jest/globals';
import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import HistoryItem from '../HistoryItem';

configure({ adapter: new Adapter() });

describe('Test HistoryItem', () => {
  it('Should match snapshot', () => {
    const component = shallow(<HistoryItem number={10} result="55" />);

    expect(toJson(component)).toMatchSnapshot();
  });
});
