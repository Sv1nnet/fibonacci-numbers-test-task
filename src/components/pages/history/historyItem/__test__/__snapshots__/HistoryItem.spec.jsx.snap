// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test HistoryItem Should match snapshot 1`] = `
<tr>
  <th>
    10
  </th>
  <th>
    55
  </th>
</tr>
`;
